extensions/amp-pinterest/0.1/util.js

```diff
@@ -39,20 +39,13 @@ function log(queryParams) {
 }
 
 /**
- * Strip data from string
+ * Pinterest provides text HTML encoded. This utility transforms it back
+ * into UTF-8 text.
  * @param {string} str - the string to filter
  * @return {string}
  */
 function filter(str) {
-  let decoded, ret;
-  decoded = '';
-  ret = '';
-  try {
-    decoded = decodeURIComponent(str);
-  } catch (e) {}
-  ret = decoded.replace(/</g, '&lt;');
-  ret = ret.replace(/>/g, '&gt;');
-  return ret;
+  return new DOMParser().parseFromString(str, 'text/html').body.textContent;
```

Comment:
Do we need to handle error conditions?

https://developer.mozilla.org/en-US/docs/Web/API/DOMParser#Error_handling

Reply:
I think that is quite unlikely. We'd see in canary, but this should run on well formed data

Reply:
But added. Not worth a round-trip :)